tools/generators/epic-generator/index.spec.ts

```diff
@@ -0,0 +1,275 @@
+import { addProjectConfiguration, ProjectType, serializeJson, stripIndents } from '@nrwl/devkit';
+import { createTreeWithEmptyWorkspace } from '@nrwl/devkit/testing';
+import { execSync } from 'child_process';
+import epicGenerator from './index';
+import { readFileSync } from 'fs';
+
+jest.mock('child_process');
+// @ts-ignore
+const execSyncMock = <jest.Mock<string>>execSync;
+
+jest.mock('fs');
+// @ts-ignore
+const readFileSyncMock = <jest.Mock<string>>readFileSync;
+
+type Package = {
+  name: string;
+  version: string;
+  projectType: ProjectType;
+  owners: string[];
+};
+function setupTest(packages: Package[]) {
+  const tree = createTreeWithEmptyWorkspace();
+
+  // Initialize NX package structure
+  packages.forEach(pckg => {
+    // package.json initialization
+    tree.write(
+      `packages/${pckg.name}/package.json`,
+      serializeJson({
+        name: pckg.name,
+        version: pckg.version,
+      }),
+    );
+
+    // workspace.json initialization
+    addProjectConfiguration(tree, pckg.name, {
+      root: `packages/${pckg.name}`,
+      projectType: pckg.projectType,
+    });
+  });
+
+  // set mock CODEOWNERS response for readFileSync
+  const codeownersContent = packages
+    .map(pckg => {
+      const rootPath = pckg.projectType == 'application' ? 'apps' : 'packages';
+
+      return `${rootPath}/${pckg.name} ${pckg.owners.join(' ')}`;
+    })
+    .join('\n');
+  readFileSyncMock.mockReturnValueOnce(codeownersContent);
+
+  // set execSync mock calls
+  execSyncMock.mockReturnValueOnce('Logged in to github.com.'); // response to 'gh auth'
+  execSyncMock.mockReturnValueOnce('epicUrl'); // response to epic creation
+
+  /**
+   * Responses for each of the packages created
+   * Mimics implementation and adds an "unknown" owner to mock
+   * for packages with no owner
+   * Only accepts teams as owners (discards owners with no '/')
+   */
+  packages
+    .filter(pckg => pckg.projectType == 'library')
+    .flatMap(pckg => (pckg.owners.length > 0 ? pckg.owners : ['unknown']))
+    .reduce<string[]>((acc, owner) => {
+      if (owner.includes('/') && !acc.includes(owner)) {
+        acc.push(owner);
+      }
+
+      return acc;
+    }, [])
+    .forEach(owner => {
+      execSyncMock.mockReturnValueOnce(`issueUrl-${owner}`);
+    });
+
+  execSyncMock.mockReturnValueOnce('epicUrl'); // response to editing the epic
+
+  return tree;
+}
+
+describe('epic-generator', () => {
+  describe('validation', () => {
+    it('requires a title', () => {
+      const tree = createTreeWithEmptyWorkspace();
+
+      expect(() => epicGenerator(tree, {} as any)).toThrowError('Must provide a title for the issue');
+    });
+
+    it('requires a non-empty title', () => {
+      const tree = createTreeWithEmptyWorkspace();
+
+      expect(() => epicGenerator(tree, { title: ' ' })).toThrowError('Must provide a title for the issue');
```

Comment:
consider using `toThrowErrorMatchingInlineSnapshot`

this is a brittle test that would require more effort from dev to be updated